.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,58 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-dockerfile
+{
+  "name": "MatX Development Container",
+  "build": {
+    // Sets the run context to one level up instead of the .devcontainer folder.
+    "context": "..",
+    // Update the 'dockerFile' property to correct location of production Dockerfile
+    "dockerfile": "./dev.Dockerfile",
+    "args": {
+      "REMOTE_USER": "${localEnv:USER}",
+      "REMOTE_UID": "${localEnv:REMOTE_UID:1000}",
+      "REMOTE_GID": "${localEnv:REMOTE_GID:1000}"
+    }
+  },
+  "privileged": true,
+  "capAdd": [
+    "SYS_NICE",
+    "SYS_PTRACE"
+  ],
+  "hostRequirements": {
+    "gpu": "optional"
+  },
+  "securityOpt": [
+    "seccomp=unconfined"
+  ],
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  // "features": {},
+  // Use 'forwardPorts' to make a list of ports inside the container available locally.
+  "forwardPorts": [
+    3000,
```

Comment:
Why do we need 3000 and 6006?

Reply:
this may be residual as well. I can remove them if we don't have a compelling reason for any ports in the dev image. I am creating a different DLI image which will have the ports opened for the notebooks.... do we want to make the standard dev container also have ports open? (I don't have a strong opinion here)

Reply:
I think 8888 is fine since we want Jupyter exposed, but the rest we don't need